### ISSUE TITLE
fix(external-link): set custom User-Agent and add allowedStatuses config option

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#155]
## Overview

The \`external-link\` rule incorrectly reports many valid links as broken. The root cause is twofold:

1. Go's default \`User-Agent: Go-http-client/1.1\` is blocked by CDN-level bot protection (Cloudflare, Akamai) used by Wikipedia, academic journal publishers, and others — returning 403 before the request reaches the content server.
2. There is currently no way for users to configure which HTTP status codes should be treated as non-errors, leaving 403 responses indistinguishable from genuine link failures.

Reported in #147.

## Problem

```sh
# Wikipedia returns 403 to Go's default UA
curl -s -o /dev/null -w "%{http_code}\n" \
  -A "Go-http-client/1.1" \
  https://en.wikipedia.org/wiki/Quantitative_trait_locus
# → 403

# doi.org redirects to publisher sites that also return 403
curl -s -o /dev/null -w "%{http_code}\n" \
  -L -A "Go-http-client/1.1" \
  https://doi.org/10.1086/695704
# → 403 (final: journals.uchicago.edu)

# A descriptive UA fixes Wikipedia
curl -s -o /dev/null -w "%{http_code}\n" \
  -A "gomarklint/v2 (+https://github.com/shinagawa-web/gomarklint)" \
  https://en.wikipedia.org/wiki/Quantitative_trait_locus
# → 200
```

### 1. Set a descriptive User-Agent header

Send \`User-Agent: gomarklint/v2 (+https://github.com/shinagawa-web/gomarklint)\` on all HTTP requests made by \`performCheck\`. This resolves 403s from sites that specifically block \`Go-http-client/1.1\` (e.g. Wikipedia).

### 2. Add \`allowedStatuses\` config option

Some sites (major journal publishers, social media, etc.) return 403 regardless of User-Agent — bot protection at the CDN edge is unavoidable. A new \`allowedStatuses\` option lets users opt out of treating specific status codes as errors.

```json
"external-link": {
  "enabled": true,
  "severity": "error",
  "timeoutSeconds": 5,
  "skipPatterns": [],
  "allowedStatuses": [403]
}
```

### 3. Update docs: limitations of external link checking

External link checking has inherent limitations that users should understand:

- Sites protected by Cloudflare, Akamai, or similar CDNs may return 403 to any automated client regardless of User-Agent. This is expected behavior, not a bug in the link itself.
- **Particularly strict categories:**
  - **Academic publishers** (doi.org destinations): Elsevier, Wiley, Springer/Nature, APS, PNAS, journals.uchicago.edu, etc.
  - **Reference sites**: Wikipedia (mitigated by UA fix), ResearchGate
  - **Social media**: Twitter/X, LinkedIn, Facebook
  - **Search engines**: Google, Bing
- Users should use \`skipPatterns\` to exclude known-strict domains, or \`allowedStatuses: [403]\` to treat access-denied responses as acceptable. The tool can flag candidates, but the final judgment on link validity belongs to the author.

## References

- #147 — User report

**Comment by shinagawa-web** [comment]
## Implementation Plan

### Files to create

- \`docs/external-link-limitations.md\` — standalone doc on the limitations of external link checking (bot protection, recommended workarounds, list of known-strict sites)

### Files to modify

- \`internal/rule/external_link.go\` — add \`User-Agent\` header to HEAD and GET requests in \`performCheck\`; honor \`allowedStatuses\` in \`CheckExternalLinks\`
- \`internal/config/config.go\` — add \`AllowedStatuses []int\` field to \`ExternalLinkConfig\`
- \`internal/config/config_test.go\` — cover new field
- \`internal/rule/external_link_test.go\` — cover UA header and \`allowedStatuses\` behavior
- \`testdata/.gomarklint.example.json\` — add \`allowedStatuses\` example
- \`README.md\` / \`README.ja.md\` — note on external link checking limitations and link to new doc

### Checklist

- [ ] Set \`User-Agent: gomarklint/v2 (+https://github.com/shinagawa-web/gomarklint)\` in \`performCheck\` (HEAD and GET)
- [ ] Add \`AllowedStatuses []int\` to \`ExternalLinkConfig\` with default \`[]\`
- [ ] Skip reporting when response status is in \`allowedStatuses\`
- [ ] Unit tests: UA header sent; 403 skipped when in \`allowedStatuses\`; 403 reported when not in \`allowedStatuses\`
- [ ] Update example config with \`allowedStatuses\`
- [ ] Add \`docs/external-link-limitations.md\` with bot-protection context and known-strict site list
- [ ] Add note + link in README (both EN and JA)
- [ ] \`make test-all\` passes
